version7.go

```diff
@@ -82,14 +93,14 @@ var lastV7time int64
 
 const nanoPerMilli = 1000000
 
-// getV7Time returns the time in milliseconds and nanoseconds / 256.
+// getV7TimeFromTime returns the time in milliseconds and nanoseconds / 256 for a custom time.
 // The returned (milli << 12 + seq) is guaranteed to be greater than
 // (milli << 12 + seq) returned by any previous call to getV7Time.
-func getV7Time() (milli, seq int64) {
+func getV7TimeFromTime(t time.Time) (milli, seq int64) {
 	timeMu.Lock()
 	defer timeMu.Unlock()
 
-	nano := timeNow().UnixNano()
+	nano := t.UnixNano()
```

Comment:
Due to the behavior later in this function with `lastV7time` I don't think we want that when using a custom timestamp. Basically if you pass in a future time, then a past time, the past time will get reset to be in the future. I opened https://github.com/google/uuid/pull/182 that doesn't have this issue.